kbcstorage/base.py

```diff
@@ -0,0 +1,139 @@
+"""
+Base classes for constructing the client.
+
+Primarily exposes a base Endpoint class which deduplicates functionality across
+various endpoints, such as tables, workspaces, jobs, etc. as described in the
+`Storage API documentation`.
+
+
+.. _Storage API documentation:
+    http://docs.keboola.apiary.io/
+"""
+from urllib.parse import urljoin
+
+import requests
+
+
+class Endpoint:
+    """
+    Base class for implementing a single endpoint related to a single entities
+    as described in the Storage API.
+
+    Attributes:
+        path (str): URL for this endpoint.
+        token (str): A key for the Storage API.
+    """
+    def __init__(self, root, extension, token):
+        """
+        Create an endpoint.
+
+        Args
+            root (str): Root url of API. eg.
+                "https://connection.keboola.com/v2/storage/"
+            extension (str): Extension of url for the endpoint. eg. "buckets"
+            token (str): A key for the Storage API. Can be found in the storage
+            console.
+        """
+        self.path = urljoin(root, extension)
+        self.token = token
+
+    def _get(self, params=[], extra_headers={}):
+        """
+        Make a get request to the url of the endpoint extended with additional
+        params.
+
+        Args:
+            params (:obj:`list`): Is used to update the url of the request.
+                Default [].
+            extra_headers (:obj:`dict`): Is used to update the headers.
+                Default {}.
+
+        Returns:
+            body: Response body parsed from json.
+
+        Raises:
+            requests.HTTPError: If the API request fails.
+        """
+        headers = {'X-StorageApi-Token': self.token}
+        headers.update(extra_headers)
+
+        url = self._extend(self.path, params)
+
+        r = requests.get(url, headers=headers)
+        r.raise_for_status()
+
+        return r.json()
+
+    def _post(self, body={}, params=[], extra_headers={}):
```

Comment:
`requests` can handle array parameters, but it encodes the `[]` into `%5B%5D`. That should not be an issue for the storage api, right?

try this:
```
>>> import requests
>>> r =requests.get('https://httpbin.org/', params={'whereValues[]':['12','34','56']})
>>> r.url
'https://httpbin.org/?whereValues%5B%5D=12&whereValues%5B%5D=34&whereValues%5B%5D=56'
>>>
```

So really all we need to do is to expose the native `requests` functionality in the `Endpoint`s  get method and we should be good to go